src/client/mod.rs

```diff
@@ -31,10 +33,15 @@ impl<T> Grpc<T> {
         Grpc { inner }
     }
 
-    pub fn poll_ready<R>(&mut self) -> Poll<(), ::Error<T::Error>>
-    where T: HttpService<R> {
+    pub fn poll_ready<R>(&mut self) -> Poll<(), ::Status>
+    where
+        T: HttpService<R>,
+        T::Error: Error + 'static,
```

Comment:
This bound will not work with any service that has an error of type `Box<Error>` I believe.